### PR TITLE
Cache-control improvements

### PR DESCRIPTION
**Pull request by msztorc**
- added optional `$cache` arg for `execute` method
- added `clearCache()` method
- added `getCache()` method
- added some test assertions regarding above changes

### CODE REVIEW

**Comment by phpfui** [comment]
Awsome!  Will look at this tonight.  Thanks for the PR!

File: src/NXP/MathExecutor.php [hunk 1]
```diff
@@ -412,22 +412,25 @@ protected function defaultFunctions() : array
...
      */
-    public function execute(string $expression)
+    public function execute(string $expression, bool $cache = true)
     {
         $cachekey = $expression;
         if (!array_key_exists($cachekey, $this->cache)) {
             $tokens = (new Tokenizer($expression, $this->operators))->tokenize()->buildReversePolishNotation();
-            $this->cache[$cachekey] = $tokens;
+            if ($cache)
+                $this->cache[$cachekey] = $tokens;
```
Comment (neonxp) [inline src/NXP/MathExecutor.php:429]:
```suggestion
            if ($cache) {
                $this->cache[$cachekey] = $tokens;
            }
```

**Review by neonxp — APPROVED** [review]
Looks good 👍🏻

**Comment by phpfui** [comment]
This all looks good.  Thanks for the contribution.

**Pull request merged**